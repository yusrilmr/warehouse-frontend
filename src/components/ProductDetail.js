/* eslint-disable */
import React, { useState } from 'react';
import ReactTable from "react-table";
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import { SERVER_URL } from "../constants";

const ProductDetail = (props) => {
    const [open, setOpen] = useState(false);
    const [productDetail, setProductDetail] = useState([]);
    const [product, setProduct] = useState({
        id: '', name: ''
    });

    const fetchProductDetail = (productId) => {
        const token = sessionStorage.getItem("jwt");
        fetch(SERVER_URL + 'product-details/' + productId,
            {
                headers: {'Authorization': token}
            })
            .then((response) => response.json())
            .then((responseData) => {
                setProductDetail(responseData);
            })
            .catch(err => console.error(err));
    }

    const handleClickOpen = () => {
        console.log(props.product);
        setProduct({
            id: props.product.productId, name: props.product.productName
        })
        fetchProductDetail(props.product.productId);
        setOpen(true);
    };

    const handleClose = () => {
        setOpen(false);
    };

    const columns = [
        { Header: 'Article id', accessor: 'identification'},
        { Header: 'Article Name', accessor: 'name'},
        { Header: 'Amount', accessor: 'totalArticle'},
        { Header: 'Stock', accessor: 'stock'},
    ];
    return (
        <div>
            <Button color="primary" size="small" onClick={handleClickOpen}>Detail</Button>
            <Dialog fullWidth maxWidth="sm" open={open} onClose={handleClose}>
                <DialogTitle>Product Detail</DialogTitle>
                <DialogContent style={{height:'500px'}}> {/* the max height already handled by material-ui */}
                    <ReactTable data={productDetail} columns={columns} filterable={true} defaultPageSize= {10}/>
                </DialogContent>
                <DialogActions>
                    <Button color="secondary" onClick={handleClose}>Close</Button>
                </DialogActions>
            </Dialog>
        </div>
    );
};
export default ProductDetail;