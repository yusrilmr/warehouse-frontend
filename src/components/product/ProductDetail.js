/* eslint-disable */
import React, { useState } from 'react';
import ReactTable from "react-table";
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import ProductAPI from "../../services/productAPI";

const ProductDetail = (props) => {
    const [open, setOpen] = useState(false);
    const [productDetail, setProductDetail] = useState([]);
    const [product, setProduct] = useState({ id: '', name: '' });

    const refreshProductDetail = (productId) => {
        new ProductAPI().fetchProductDetail(productId)
            .then((responseData) => {
                setProductDetail(responseData);
            })
            .catch(err => console.error(err));
    }

    const handleClickOpen = () => {
        setProduct({
            id: props.product.productId, name: props.product.productName
        })
        refreshProductDetail(props.product.productId);
        setOpen(true);
    };

    const handleClose = () => {
        setOpen(false);
    };

    const columns = [
        { Header: 'Article id', accessor: 'articleIdentification'},
        { Header: 'Article Name', accessor: 'articleName'},
        { Header: 'Amount', accessor: 'totalArticle'},
        { Header: 'Stock', accessor: 'stock'},
    ];
    return (
        <div>
            <Button color="primary" size="small" onClick={handleClickOpen}>Detail</Button>
            <Dialog fullWidth maxWidth="sm" open={open} onClose={handleClose}>
                <DialogTitle>Product Detail</DialogTitle>
                <DialogContent style={{height:'500px'}}> {/* the max height already handled by material-ui */}
                    <ReactTable data={productDetail} columns={columns} filterable={true} defaultPageSize= {10}/>
                </DialogContent>
                <DialogActions>
                    <Button color="secondary" onClick={handleClose}>Close</Button>
                </DialogActions>
            </Dialog>
        </div>
    );
};
export default ProductDetail;