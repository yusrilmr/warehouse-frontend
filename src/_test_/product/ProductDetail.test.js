import React from 'react';
import ReactTable from "react-table";
import ProductDetail from '../../components/product/ProductDetail';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import TestRenderer from "react-test-renderer";

Enzyme.configure({ adapter: new Adapter() });

describe('<ProductDetail />', () => {
    it('renders one <ReactTable /> components', () => {
        const wrapper = shallow(<ProductDetail />);
        expect(wrapper.find(ReactTable)).toHaveLength(1);
    });
});

it('renders a snapshot', () => {
    const tree = TestRenderer.create(<ProductDetail/>).toJSON();
    expect(tree).toMatchSnapshot();
});