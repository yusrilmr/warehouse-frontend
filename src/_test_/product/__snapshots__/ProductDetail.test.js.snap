// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Detail
    </span>
  </button>
</div>
`;
